lib/stm32/common/desig_common_all.c

```diff
@@ -19,35 +19,22 @@
 
 #include <libopencm3/stm32/desig.h>
 
-uint16_t desig_get_flash_size(void)
-{
-	return DESIG_FLASH_SIZE;
-}
-
-void desig_get_unique_id(uint32_t *result)
-{
-	*result++ = DESIG_UNIQUE_ID2;
-	*result++ = DESIG_UNIQUE_ID1;
-	*result = DESIG_UNIQUE_ID0;
-}
-
-void desig_get_unique_id_as_string(char *string,
-				   unsigned int string_len)
+void desig_get_unique_id_as_string(char *string, unsigned int string_len)
 {
 	int i, len;
-	uint32_t dev_id_buf[3];
-	uint8_t *device_id = (uint8_t *)dev_id_buf;
+	uint32_t uid_buf[3];
+	uint8_t *uid = (uint8_t *)uid_buf;
 	const char chars[] = "0123456789ABCDEF";
 
-	desig_get_unique_id(dev_id_buf);
+	desig_get_unique_id(uid_buf);
 
 	/* Each byte produces two characters */
-	len = (2 * sizeof(dev_id_buf) < string_len) ?
-		2 * sizeof(dev_id_buf) : string_len - 1;
+	len = (2 * sizeof(uid_buf) < string_len) ?
+		2 * sizeof(uid_buf) : string_len - 1;
 
 	for (i = 0; i < len; i += 2) {
-		string[i]     = chars[(device_id[i / 2] >> 4) & 0x0F];
-		string[i + 1] = chars[(device_id[i / 2] >> 0) & 0x0F];
+		string[i]     = chars[(uid[i / 2] >> 4) & 0x0F];
+		string[i + 1] = chars[(uid[i / 2] >> 0) & 0x0F];
```

Comment:
this is all gratuitous in this function, you can just drop it.

Reply:
I thought it would be nice to change the names here, since "device ID" actually means something else (the device ID field in DBGMCU). I feel it would be especially confusing to anyone tracing this code now since that other device ID is actually used to decide the address now. I can revert them if you want, though. Is there a benefit to not changing this? It's an internal name that no one should really care about?

Reply:
yeah, I mean, whatever, it's just generally bad form to make unrelated changes.  it makes the diffs much bigger and harder to review, and we end up wasting time on this sort of cruft instead of the issues itself.

Reply:
Ah I had these changes in a separate commit before, but squashed them for the PR. Will keep them separate next time.